homeassistant/components/webhook/trigger.py

```diff
@@ -2,24 +2,45 @@
 from __future__ import annotations
 
 from dataclasses import dataclass
+import logging
 
 from aiohttp import hdrs
 import voluptuous as vol
 
 from homeassistant.const import CONF_PLATFORM, CONF_WEBHOOK_ID
 from homeassistant.core import CALLBACK_TYPE, HassJob, HomeAssistant, callback
 import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.issue_registry import (
+    IssueSeverity,
+    async_create_issue,
+    async_delete_issue,
+)
 from homeassistant.helpers.trigger import TriggerActionType, TriggerInfo
 from homeassistant.helpers.typing import ConfigType
 
-from . import DOMAIN, async_register, async_unregister
+from . import (
+    DEFAULT_METHODS,
+    DOMAIN,
+    SUPPORTED_METHODS,
+    async_register,
+    async_unregister,
+)
+
+_LOGGER = logging.getLogger(__name__)
 
 DEPENDENCIES = ("webhook",)
 
+CONF_ALLOWED_METHODS = "allowed_methods"
+CONF_LOCAL_ONLY = "local_only"
+
 TRIGGER_SCHEMA = cv.TRIGGER_BASE_SCHEMA.extend(
     {
         vol.Required(CONF_PLATFORM): "webhook",
         vol.Required(CONF_WEBHOOK_ID): cv.string,
+        vol.Optional(CONF_ALLOWED_METHODS): vol.All(
+            cv.ensure_list, [vol.In(SUPPORTED_METHODS)], vol.Unique()
```

Comment:
We can handle the `upper()` part here:
```suggestion
            cv.ensure_list, [vol.All(vol.Upper, vol.In(SUPPORTED_METHODS))], vol.Unique()
```